### PR TITLE
Notebook controller notebooks backend.

### PR DESCRIPTION
**Pull request by maroroman**
This PR adds GET, POST, PATCH and DELETE calls for notebooks. 

It also renames the notebook images part of the dashboard to be compatible.

Related to: https://github.com/opendatahub-io/odh-dashboard/issues/148

### CODE REVIEW

**Comment by lucferbux** [comment]
There are a few lints error, once they're fixed it's a lgtm for me

File: backend/src/types.ts [hunk 1]
```diff
@@ -245,67 +244,152 @@ export type BuildStatus = {
   timestamp?: string;
 };
 
+export type EnvironmentVariable = {
```
Comment (dlabaj) [inline backend/src/types.ts:247]: types and rest calls should be added to the api documentation from this PR. https://github.com/opendatahub-io/odh-dashboard/pull/246. once it goes in.  I would create a new issue for that and do it in another PR.

**Review by dlabaj — APPROVED** [review]
lgtm just a comment about creating a new bug/issue to document the api later.

File: backend/src/routes/api/notebook-images/notebooksImageStreamUtils.ts [hunk 2]
```diff
@@ -39,20 +39,20 @@ const mapImageStreamToNotebook = (is: ImageStreamKind): Notebook => ({
...
     (JSON.parse(
       is.spec.tags[0].annotations['opendatahub.io/notebook-software'],
-    ) as NotebookPackage[]),
+    ) as NotebookImagePackage[]),
   uploaded: is.metadata.creationTimestamp,
   url: is.metadata.annotations['opendatahub.io/notebook-image-url'],
   user: is.metadata.annotations['opendatahub.io/notebook-image-creator'],
 });
 
 export const getNotebooks = async (
```
Comment (DaoDaoNoCode) [inline backend/src/routes/api/notebook-images/notebooksImageStreamUtils.ts:53]: In this function, you are only getting image streams with `app.kubernetes.io/created-by=byon` label. Should we get all the image streams for KFNBC? Or we can set a query param to differentiate that?

File: backend/src/types.ts (hunk 1 above)
Comment (maroroman) [reply backend/src/types.ts:247]: Will do 👍

File: backend/src/routes/api/notebook-images/notebooksImageStreamUtils.ts (hunk 2 above)
Comment (maroroman) [reply backend/src/routes/api/notebook-images/notebooksImageStreamUtils.ts:53]: Right, we should definitely differentiate that, I will try to make some logic around it.

Comment (DaoDaoNoCode) [reply backend/src/routes/api/notebook-images/notebooksImageStreamUtils.ts:53]: @maroroman BTW, could you please also add the `tags` to the return object? For BYON, it may only use packages and software so it's OK to only return `is.spec.tags[0]`, but for KFNBC it should return all the tags, and users could choose different tags of the same image stream.

Comment (cfchase) [reply backend/src/routes/api/notebook-images/notebooksImageStreamUtils.ts:53]: After talking to @DaoDaoNoCode it looks like this api call is customized heavily for byon.  I suggested he create his own endpoint and work with @maroroman to review and modify as needed.

**Review by cfchase — APPROVED** [review]

**Review by dlabaj — APPROVED** [review]
